components/Layout.tsx

```diff
@@ -1,15 +1,33 @@
-import React from 'react'
+import React, {useEffect} from 'react'
 import Header from './Header/Header'
+import {useUser} from "@auth0/nextjs-auth0";
+import {userProfileState} from "../atoms/userProfile";
+import {useSetRecoilState} from "recoil";
+import {getUserFromFirebase} from "../utils/helpers/userHelper";
 
 interface LayoutProps {
     children: React.ReactNode
 }
 
-const Layout = ({ children }: LayoutProps) => {
+
+const Layout = ({children}: LayoutProps) => {
+    const {user, isLoading} = useUser();
+    const setUser = useSetRecoilState(userProfileState);
+
+    useEffect(() => {
+        if (user) {
```

Comment:
Please, add comments on what these functions are supposed to do and how they work.